### DISCUSSION TITLE
highlighting rust language in a react component

### DISCUSSION THREAD

**Discussion by arunkumar413** [PrismJS/prism#3286] [Q&A]
Hi Team,

Could you please  provide an example to highlight rust language in a react component. Below is the process I followed but the code highlighting didn't work.

1) Installed prism
2) installed the babel plugin
3) created the `.babelrc` file

```
{
  "plugins": [
    ["prismjs", {
        "languages": ["javascript", "css", "markup", "rust"],
        "plugins": ["line-numbers"],
        "theme": "twilight",
        "css": true
    }]
  ]
}
```

```
import Prism from 'prismjs';
import "prismjs/themes/prism-tomorrow.css";

    <pre className="line-numbers">
          <code className="language-rust">
                    {my rust code here}
          </code>
     </pre>

```

**Comment by RunDevelopment** [comment]
Please see [this tutorial](https://betterstack.dev/blog/code-highlighting-in-react-using-prismjs/) (linked on our website). It includes an example of how to do this.

I hope this helps.

**Reply by arunkumar413** [reply]
@RunDevelopment No, it didn't help. That's why I raised this ticket. Not sure what made you think that I didn't follow that tutorial link before raising this ticket.

**Reply by RunDevelopment** [reply]
Well, the code in the tutorial should work, so maybe there's something wrong with your setup. Could you please provide a minimal example project that reproduces your problem?

**Comment by arunkumar413** [comment]
Here is my the code that isn't working. 

https://codesandbox.io/s/crimson-darkness-omj3w?file=/src/pages/Button.js

Open the output and select Rust language from the drop down at the top-right.

**Reply by mAAdhaTTah** [reply]
Create React App doesn't support modifying babel, so the babelrc doesn't do anything. You'll need to either eject or find some other way to modify the babel configuration (e.g. craco).
